contrib/diurnal/dc_test.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"testing"
+	"time"
+)
+
+func equalsTimeCounts(a, b []timeCount) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i := range a {
+		if a[i].time != b[i].time || a[i].count != b[i].count {
+			return false
+		}
+	}
+	return true
+}
+
+func TestParseTimeCounts(t *testing.T) {
+	cases := []struct {
+		times  string
+		counts string
+		out    []timeCount
+		err    bool
+	}{
+		{
+			"1s,2m,3h,4h", "1,4,1,8", []timeCount{
```

Comment:
I would expect these to be absolute time-of-day values, rather than durations - are the cumulative or absolute?